lager/detail/smart_lens.hpp

```diff
@@ -27,9 +29,19 @@ class flex_vector;
 namespace lager {
 namespace detail {
 
+template <typename T, typename Key>
+using at_t = std::decay_t<decltype(std::declval<T>().at(std::declval<Key>()))>;
+
+template <typename T>
 struct smart_lens_base
 {
-    template <typename Key>
+    template <typename Lens, std::enable_if_t<!zug::meta::is_detected<at_t, T, Lens>::value, int> = 0>
```

Comment:
I would prefer to detect this case by using a `is_detected` expression that explicitly checks if `Lens` supports `set(l, declval<T>(), view(l, declval<T>()))`.

Reply:
so the problem here is that `set`, `view`, all the lenses, and just about everything we're using are all using return type deduction, [which unfortunately _doesn't_ work for SFINAE]...(https://en.cppreference.com/w/cpp/language/function#Return_type_deduction)
for that reason, I decided to switch on the `at_t` instead.